### PR TITLE
osutil/disks: add DiskFromName to get a disk using a udev name

### PR DESCRIPTION
**Pull request by anonymouse64**
This will be needed from the lk bootloader where the bootloader will tell us the
name of a disk we should trust for finding specifically named partitions.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Looks good, thank you!

File: osutil/disks/disks.go [hunk 1]
```diff
@@ -145,6 +145,41 @@ func parseUdevProperties(r io.Reader) (map[string]string, error) {
 	return m, scanner.Err()
 }
 
+// DiskFromName finds a matching Disk using the specified name, such as vda, or
+// mmcblk0, etc.
+func DiskFromName(name string) (Disk, error) {
```
Comment (bboozzoo) [inline osutil/disks/disks.go:150]: s/name/deviceName/  then?

Comment (anonymouse64) [inline osutil/disks/disks.go:150]:
```suggestion
func DiskFromDeviceName(deviceName string) (Disk, error) {
```

File: osutil/disks/disks.go [hunk 2]
```diff
@@ -145,6 +145,41 @@ func parseUdevProperties(r io.Reader) (map[string]string, error) {
...
+	// query for the disk props using udev
+	props, err := udevProperties(name)
```
Comment (anonymouse64) [inline osutil/disks/disks.go:152]:
```suggestion
	props, err := udevProperties(deviceName)
```

File: osutil/disks/disks.go [hunk 3]
```diff
@@ -145,6 +145,41 @@ func parseUdevProperties(r io.Reader) (map[string]string, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	major, err := strconv.Atoi(props["MAJOR"])
+	if err != nil {
+		return nil, fmt.Errorf("cannot find disk with name %q: malformed udev output", name)
```
Comment (anonymouse64) [inline osutil/disks/disks.go:159]:
```suggestion
		return nil, fmt.Errorf("cannot find disk with name %q: malformed udev output", deviceName)
```

File: osutil/disks/disks.go [hunk 4]
```diff
@@ -145,6 +145,41 @@ func parseUdevProperties(r io.Reader) (map[string]string, error) {
...
+	}
+	minor, err := strconv.Atoi(props["MINOR"])
+	if err != nil {
+		return nil, fmt.Errorf("cannot find disk with name %q: malformed udev output", name)
```
Comment (anonymouse64) [inline osutil/disks/disks.go:163]:
```suggestion
		return nil, fmt.Errorf("cannot find disk with name %q: malformed udev output", deviceName)
```

File: osutil/disks/disks.go [hunk 5]
```diff
@@ -145,6 +145,41 @@ func parseUdevProperties(r io.Reader) (map[string]string, error) {
...
+	}
+
+	// ensure that the device has DEVTYPE=disk, if not then we were not given a
+	// disk name
+	devType := props["DEVTYPE"]
+	if devType != "disk" {
+		return nil, fmt.Errorf("device %q is not a disk, it has DEVTYPE of %q", name, devType)
```
Comment (anonymouse64) [inline osutil/disks/disks.go:170]:
```suggestion
		return nil, fmt.Errorf("device %q is not a disk, it has DEVTYPE of %q", deviceName, devType)
```

File: osutil/disks/disks_test.go [hunk 6]
```diff
@@ -89,6 +89,52 @@ func (s *diskSuite) SetUpTest(c *C) {
...
+		c.Assert(dev, Equals, "sda")
+		return map[string]string{
+			"MAJOR":   "1",
+			"MINOR":   "2",
+			"DEVTYPE": "disk",
+		}, nil
+	})
+	defer restore()
+
+	d, err := disks.DiskFromName("sda")
```
Comment (anonymouse64) [inline osutil/disks/disks_test.go:103]:
```suggestion
	d, err := disks.DiskFromDeviceName("sda")
```

File: osutil/disks/disks_test.go [hunk 7]
```diff
@@ -89,6 +89,52 @@ func (s *diskSuite) SetUpTest(c *C) {
...
+		c.Assert(dev, Equals, "sda1")
+		return map[string]string{
+			"MAJOR":   "1",
+			"MINOR":   "3",
+			"DEVTYPE": "partition",
+		}, nil
+	})
+	defer restore()
+
+	_, err := disks.DiskFromName("sda1")
```
Comment (anonymouse64) [inline osutil/disks/disks_test.go:119]:
```suggestion
	_, err := disks.DiskFromDeviceName("sda1")
```

File: osutil/disks/disks_test.go [hunk 8]
```diff
@@ -89,6 +89,52 @@ func (s *diskSuite) SetUpTest(c *C) {
...
+		c.Assert(dev, Equals, "sda")
+		// udev should always return the major/minor but if it doesn't we should
+		// fail
+		return map[string]string{
+			"MAJOR": "blah blah blah",
+		}, nil
+	})
+	defer restore()
+
+	_, err := disks.DiskFromName("sda")
```
Comment (anonymouse64) [inline osutil/disks/disks_test.go:134]:
```suggestion
	_, err := disks.DiskFromDeviceName("sda")
```

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**